db.go

```diff
@@ -795,15 +801,35 @@ func (db *DB) buildBTreeIdx(r *Record) {
 		return
 	}
 
-	bucket, key := r.Bucket, r.H.Key
+	bucket, key, meta := r.Bucket, r.H.Key, r.H.Meta
 
 	if _, ok := db.BTreeIdx[bucket]; !ok {
 		db.BTreeIdx[bucket] = NewBTree()
 	}
 
-	if r.H.Meta.Flag == DataDeleteFlag {
+	if meta.Flag == DataDeleteFlag {
+		db.tm.del(bucket, string(key))
 		db.BTreeIdx[bucket].Delete(key)
 	} else {
+		if meta.TTL != Persistent {
+			expireTime := time.Unix(int64(meta.Timestamp)+int64(meta.TTL), 0)
+			expire := time.Until(expireTime)
+
+			db.tm.add(bucket, string(key), expire, func() {
```

Comment:
The callback function can be a variable that define above. That will look more elegant.